src/convert.rs

```diff
@@ -0,0 +1,7 @@
+//! Internal conversion traits.
+
+/// Conversion trait that mirrors [`std::convert::From`] but is used when the `std` trait cannot be.
+pub trait From<T> {
+    /// Create an instance of this type from another type `T`.
+    fn from(value: T) -> Self;
```

Comment:
`std` impl has `Size` bound: https://doc.rust-lang.org/std/convert/trait.From.html. But I believe that is implicit already?